packages/snjs/lib/services/settings_service/SNSettingsService.ts

```diff
@@ -0,0 +1,29 @@
+import { PureService } from '../pure_service';
+
+import { SNApiService } from '../api/api_service';
+import { SNStorageService } from '../storage_service';
+import { SettingsProvider } from './SettingsProvider';
+import { StorageKey } from '@Lib/storage_keys';
+import { User } from '../api/responses';
+import { SettingsRepo } from './SettingsRepo';
+
+export class SNSettingsService extends PureService {
+  private readonly _provider: SettingsProvider;
+  private readonly getUser: () => User | null;
+
+  constructor(storageService: SNStorageService, apiService: SNApiService) {
+    super();
+    this.getUser = (): User | null =>
```

Comment:
I don't think you can do this inside the constructor because storageService may not yet be ready to read values on construction. There are stages to application initialization and storage service loading is async.

Reply:
I saw that pattern in other services, but just wanted to try if this would work - found that as you said, the services have to respect the application lifecycle :)
Fixed now. Resolving comment.